covid/utils.py

```diff
@@ -0,0 +1,63 @@
+from pathlib import Path
+import time
+import json
+import os
+
+from langdetect import detect as detect_language
+from wasabi import msg
+import pandas as pd
+
+
+SCRIPT_PATH = os.path.abspath(os.path.dirname(__file__))
+ARTICLES_DIR = os.path.join(SCRIPT_PATH, '../data/raw/CORD-19-research-challenge/')
+
+
+def get_language(text):
+    try:
+        language = detect_language(text)
+    except:
+        language = "unknown"
+    return language
+
+def yield_data(articles_dir):
+    meta_path = articles_dir + 'metadata.csv'
+    metadata = pd.read_csv(meta_path, low_memory=False)
+
+    metadata.drop_duplicates(subset=['pmcid'], inplace=True)
+    metadata.drop_duplicates(subset=['sha'], inplace=True)
+
+    metadata['publish_date'] = pd.to_datetime(metadata['publish_time'])
+    metadata = metadata[metadata['publish_date'].dt.year == 2020]
+
+    metadata['lang'] = metadata['abstract'].apply(lambda x: get_language(x))
+    metadata = metadata[metadata['lang'] == 'en']
+
+    valid_ids = set([sha.strip() for shas in metadata['sha'] for sha in shas.split(';')])
+    valid_ids.update(set(metadata['pmcid']))
```

Comment:
are we sure every paper should have a metadata entry? I'd hope so, but just flagging

Reply:
i think that this true, we can add a check in the explore data if there is none already